### ISSUE TITLE
Support using IP addresses for pin maps, location grouping/filtering

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#3159]
It would be useful to have the option to use an IP address field to place each row on a pin map instead of lat/long. Would also be neat to have the ability to use an IP address field in the query builder as a pseudo location field for the purpose of filtering or grouping.

**Comment by tlrobinson** [comment]
Yeah geocoding IPs would be great. We'll need to figure out where to get a geocoding database, and ideally how to update it since they get out of date over time.

Geocoding IPs in results for pin maps should be easy, but I suspect grouping / filtering will be harder since it needs to be done in the database/query.

**Comment by rafaelveloso** [comment]
Also note that for long term data analysis this wont be of much help. Unless you persist the geolocation on your data source at the time of request (when you first know an IP), there's no guarantees that after updating metabase's geoip database the resulting location will still be the same.

**Comment by tazumi** [comment]
+1 for that

**Comment by camsaul** [comment]
Hey everyone, one way we prioritize issues is by sorting them by the number of 👍 reactions on each issue's description. So instead of leaving `+1` or 👍  comments, please just upvote by adding a reaction to the issue description.

**Comment by JCBarry** [comment]
MaxMind support freely available GeoIP databases that can be downloaded.  We're planning to use this solution internally to create maps in Metabase.

https://dev.maxmind.com/geoip/geolite2-free-geolocation-data

**Comment by krisavi** [comment]
> MaxMind supports freely available GeoIP databases that can be downloaded. We're planning to use this solution internally to create maps in Metabase.

It is usually not so straight forward to take the database and use it for your project that you distribute. License of those providers might be the most problematic there. Probably easier solution would be to add support for it to be usable, but make user download and update database by themselves.

Ip2location has also some free version with some caveats. https://lite.ip2location.com/ip2location-lite

Metabase side it would be nice if they could just add the support, let you write database file path or take it from backend database and select "flavor" of database used (maxmind, ip2location, etc)

**Comment by abdullahdevrel** [comment]
I work for IPinfo. I think that if you need a relatively simple solution, you can use our IP API bulk endpoint. The bulk API endpoint supports hundreds of thousands of responses and can return geographic coordinates along with other location information.

Documentation: https://ipinfo.io/developers/advanced-usage

We have a free IP-to-Country database that comes with a generous CC-BY-SA 4.0 license, allowing distribution of the database (unlike most, if not all, other free IP location databases available). The data provides country-level location, which can be used as a general location indicator.

https://ipinfo.io/products/free-ip-database
